### PR TITLE
[Data Science] add python function to fetch production data

### PR DESCRIPTION
**Pull request by martindaniel4**
## What's the change

This PR adds a python utils function that returns a DataFrame with production data. 

Parameters of the function `df_production` are : 
- list of country codes
- start_date
- end_date
- delta (in minutes)

Here is an example : 

```python
import utils

data_frame = df_production(['DE','FR'],'2016-12-23','2016-12-24',1440)
```
This returns a nice data frame of production data :

```
  country  production  sources                  timestamp
0      DE         NaN      oil  2016-12-23T00:00:00+00:00
1      DE      7562.0  nuclear  2016-12-23T00:00:00+00:00
2      DE      2828.0      gas  2016-12-23T00:00:00+00:00
3      DE      4229.0  biomass  2016-12-23T00:00:00+00:00
4      DE     19919.0     coal  2016-12-23T00:00:00+00:00
5      DE         0.0    solar  2016-12-23T00:00:00+00:00
6      DE      4752.0  unknown  2016-12-23T00:00:00+00:00
7      DE     13830.0     wind  2016-12-23T00:00:00+00:00
8      DE      1381.0    hydro  2016-12-23T00:00:00+00:00
0      DE         NaN      oil  2016-12-24T00:00:00+00:00
1      DE      7289.0  nuclear  2016-12-24T00:00:00+00:00
2      DE      2149.0      gas  2016-12-24T00:00:00+00:00
3      DE      4226.0  biomass  2016-12-24T00:00:00+00:00
4      DE      9744.0     coal  2016-12-24T00:00:00+00:00
5      DE         0.0    solar  2016-12-24T00:00:00+00:00
6      DE      3412.0  unknown  2016-12-24T00:00:00+00:00
7      DE     26539.0     wind  2016-12-24T00:00:00+00:00
8      DE      1202.0    hydro  2016-12-24T00:00:00+00:00
0      FR       204.0      oil  2016-12-23T00:00:00+00:00
1      FR     48271.0  nuclear  2016-12-23T00:00:00+00:00
2      FR      8233.0      gas  2016-12-23T00:00:00+00:00
3      FR       768.0  biomass  2016-12-23T00:00:00+00:00
4      FR      1465.0     coal  2016-12-23T00:00:00+00:00
5      FR         0.0    solar  2016-12-23T00:00:00+00:00
6      FR      1508.0     wind  2016-12-23T00:00:00+00:00
```

Take a look!

### CODE REVIEW

File: datascience/utils.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import arrow
+import json
+import pandas as pd
+import requests
+
+r = requests.session()
+
+def date_range(start_date,end_date,delta):
```
Comment (corradio) [inline datascience/utils.py]: Style: use spaces between arguments.

File: datascience/utils.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    params = {
+        'countryCode': country_code,
+        'datetime': t.to('utc').isoformat()
+    }
+    obj = r.get(url, params=params).json()
+    if not obj: return
+    return obj if (t - arrow.get(obj['datetime'])).total_seconds() < delta * 60.0 else None
+
+def df_production(countries,start_date,end_date,delta):
+    df = pd.DataFrame(columns=['country','timestamp','sources','production'])
```
Comment (corradio) [inline datascience/utils.py]: Style: use spaces between arguments.

File: datascience/utils.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    time_span = date_range(start_date,end_date,delta)
```
Comment (corradio) [inline datascience/utils.py]: Style: use spaces between arguments.

File: datascience/utils.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    for country in countries:
+        print 'Fetching country %s..' % country
+        for t in time_span:
+            print 'Fetching time %s..' % t
+            o = fetch_production(country, t, delta)
+            p = pd.DataFrame({'country': country,
+                                'timestamp':t,
```
Comment (corradio) [inline datascience/utils.py]: Style: use spaces between `:` and value (same for next 2 lines).

**Comment by corradio** [comment]
Refers to #265

**Comment by corradio** [comment]
@brunolajoie you should take a quick look also! (not necessarily at the code, but at the overview @martindaniel4 have)

**Comment by corradio** [comment]
Last info @martindaniel4 : the tests are not passing because your branch is not up to date. You should merge master on it:
```
git merge master
```

**Comment by martindaniel4** [comment]
Ok will do!

**Pull request merged**